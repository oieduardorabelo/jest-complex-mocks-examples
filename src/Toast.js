import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { injectIntl, intlShape } from "react-intl";

function Toast(props) {
  return (
    <div>
      <p>Hello Toast!</p>
      <p>Toast I18N: {props.intl.formatMessage({ id: "CURRENT_DAY" })}</p>
      <b>Toast Total: {props.counter}</b>
      <button onClick={props.updateCounter}>Update!</button>
    </div>
  );
}

Toast.propTypes = {
  int: intlShape,
  updateCounter: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    counter: state.counter
  };
}

function mapDispatchToProps(dispatch) {
  return {
    updateCounter: function updateCounter() {
      dispatch({ type: "UPDATE_COUNTER" });
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Toast));
