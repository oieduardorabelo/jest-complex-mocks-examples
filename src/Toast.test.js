import React from "react";
import { render, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Toast from "./Toast";

it("renders correctly", () => {
  const actual = toJson(render(<Toast updateCounter={() => {}} />));
  expect(actual).toMatchSnapshot();
});

it("should call updateCounter correctly", () => {
  const updateCounterSpy = jest.fn();
  const actual = mount(<Toast updateCounter={updateCounterSpy} />);
  actual.find("button").simulate("click");
  expect(updateCounterSpy).toHaveBeenCalledTimes(1);
});
