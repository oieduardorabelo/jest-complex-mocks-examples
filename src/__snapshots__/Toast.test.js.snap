// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Hello Toast!
  </p>
  <p>
    Toast I18N: Today
  </p>
  <b>
    Toast Total: 
  </b>
  <button>
    Update!
  </button>
</div>
`;
